synapse/lib/ast.py

```diff
@@ -420,6 +422,14 @@ async def pivots(self, runt, node, path):
 
                 yield (pivonode, path.fork(pivonode), {'type': 'prop', 'prop': propname})
 
+            for iden in existing:
+                buid = s_common.uhex(iden)
+                othr = await node.snap.getNodeByBuid(buid)
+                for propname, ndef in othr.getNodeRefs():
+                    pivonode = await othr.snap.getNodeByNdef(ndef)
+                    if pivonode.iden() == node.iden():
```

Comment:
I think you could just check `node.ndef == ndef` rather than getting the full node here

Reply:
Aye, fixed